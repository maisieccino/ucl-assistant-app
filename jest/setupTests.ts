import '@testing-library/jest-native/extend-expect'
import '@testing-library/react-native/cleanup-after-each'
import 'react-native-gesture-handler/jestSetup'

// Silence the warning: Animated: `useNativeDriver` is not supported because
// the native animated module is missing
// https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
jest.mock(`react-native/Libraries/Animated/src/NativeAnimatedHelper`)
