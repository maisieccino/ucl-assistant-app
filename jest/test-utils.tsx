import { NavigationContainer } from '@react-navigation/native'
import { render as rtlRender, RenderAPI, act } from '@testing-library/react-native'
import React from 'react'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import configureStore from 'redux-mock-store'
import thunk from "redux-thunk"
import debounce from "../lib/debounce"
import { initialState as defaultInitialState } from '../redux/reducers'

const middlewares = [
  debounce.middleware,
  thunk,
]
const mockStore = configureStore(middlewares)

const render = async (
  ui: React.ReactElement,
  {
    initialState = defaultInitialState,
    store = mockStore(initialState) as Store,
    ...renderOptions
  } = {},
): Promise<RenderAPI> => {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <NavigationContainer>
        {children}
      </NavigationContainer>
    </Provider>
  )
  const result = rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  await act(async () => {})
  return result
}

export const waitForEventLoop = async (): Promise<void> => new Promise((resolve) => {
  const done = () => {
    jest.useFakeTimers()
    resolve()
  }
  jest.useRealTimers()
  setTimeout(done, 0)
})

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }
