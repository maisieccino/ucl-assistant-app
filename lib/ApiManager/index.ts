import PeopleController from './PeopleController'
import RoomsController from './RoomsController'
import TimetableController from './TimetableController'
import WorkspacesController from './WorkspacesController'

export default {
  people: PeopleController,
  rooms: RoomsController,
  timetable: TimetableController,
  workspaces: WorkspacesController,
}
