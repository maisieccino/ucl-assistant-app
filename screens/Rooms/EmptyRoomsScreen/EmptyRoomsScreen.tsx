import { Picker } from '@react-native-community/picker'
import { StackNavigationProp } from '@react-navigation/stack'
import React from 'react'
import {
  ActivityIndicator,
  StyleSheet,
  View,
} from 'react-native'
import { connect, ConnectedProps } from "react-redux"
import { Page } from "../../../components/Containers"
import SearchResult from "../../../components/SearchResult"
import { BodyText, CentredText } from "../../../components/Typography"
import type { AppStateType } from '../../../configureStore'
import ApiManager from "../../../lib/ApiManager"
import type { Room } from '../../../types/uclapi'
import type { RoomsNavigatorParamList } from "../RoomsNavigator"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 20,
  },
  subtitle: {
    marginBottom: 10,
  },
})

interface Props extends PropsFromRedux {
  navigation: StackNavigationProp<RoomsNavigatorParamList>,
}

interface State {
  emptyRooms: Array<Room>,
  error: Error,
  loadingEmptyRooms: boolean,
  selectedSite: string,
  sites: Array<unknown>,
}

export class EmptyRoomsScreen extends React.Component<
  Props,
  State
  > {
  static navigationOptions = {
    title: `Empty Rooms`,
  }

  constructor(props: Props) {
    super(props)
    this.state = {
      emptyRooms: [],
      error: null,
      loadingEmptyRooms: true,
      selectedSite: null,
      sites: [],
    }
  }

  componentDidMount(): void {
    this.fetchEmptyRooms()
  }

  fetchEmptyRooms = async (): Promise<void> => {
    const { token } = this.props
    try {
      const emptyRooms = await ApiManager.rooms.getEmptyRooms(token)
      const sites = Array.from(new Set(emptyRooms.map(({ siteid }) => siteid)))
        .map((siteid) => ({
          siteid,
          sitename: emptyRooms.find((r) => r.siteid === siteid).sitename,
        }))
      this.setState({
        emptyRooms,
        loadingEmptyRooms: false,
        sites,
      })
    } catch (error) {
      this.setState({ error: error.message })
    }
  }

  navigateToRoomDetail = (room: Room) => (): void => {
    const { navigation } = this.props
    navigation.navigate(`RoomsDetail`, { room })
  }

  onSelectSite = (value: string): void => {
    this.setState({ selectedSite: value })
  }

  renderEmptyRoom = (room = null): React.ReactNode => {
    if (room === null) {
      return null
    }

    const key = `${room.roomid}-${room.roomname}-`
      + `${room.siteid}-${room.classification_name}`

    return (
      <SearchResult
        key={key}
        topText={room.roomname}
        bottomText={room.classification_name}
        type="location"
        onPress={this.navigateToRoomDetail(room)}
      />
    )
  }

  renderResults = (): React.ReactNode => {
    const { emptyRooms, selectedSite } = this.state
    const matchingRooms = emptyRooms.filter(({ siteid }) => {
      if (selectedSite === null) {
        return true
      }
      return siteid === selectedSite
    })

    if (matchingRooms.length === 0) {
      return (
        <View>
          <CentredText>
            No empty rooms found üôÅ
          </CentredText>
        </View>

      )
    }
    return matchingRooms.map(this.renderEmptyRoom)
  }

  render(): React.ReactElement {
    const {
      error,
      sites,
      selectedSite,
      loadingEmptyRooms,
    } = this.state

    if (error) {
      return <BodyText>{`Could not fetch empty rooms: ${error}`}</BodyText>
    }
    if (loadingEmptyRooms) {
      return (
        <Page>
          <View style={styles.container}>
            <ActivityIndicator size="large" />
          </View>
        </Page>
      )
    }
    return (
      <Page>
        <View style={styles.container}>
          <BodyText>
            Filter by:
          </BodyText>
          <Picker
            selectedValue={selectedSite}
            onValueChange={this.onSelectSite}
            testID="building-picker"
          >
            <Picker.Item label="All Buildings" value={null} />
            {
              sites.map(({ sitename, siteid }) => (
                <Picker.Item key={siteid} label={sitename} value={siteid} />
              ))
            }
          </Picker>
          <BodyText style={styles.subtitle}>
            Rooms vacant for the next hour:
          </BodyText>
          {this.renderResults()}
        </View>
      </Page>
    )
  }
}

const connector = connect(
  (state: AppStateType) => ({
    token: state.user.token,
  }),
  () => ({}),
)

type PropsFromRedux = ConnectedProps<typeof connector>

export default connector(EmptyRoomsScreen)
