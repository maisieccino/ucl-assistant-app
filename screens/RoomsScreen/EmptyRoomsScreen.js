import PropTypes from "prop-types"
import React from 'react'
import {
  ActivityIndicator,
  Picker,
  StyleSheet,
  View,
} from 'react-native'
import { connect } from "react-redux"

import { Page } from "../../components/Containers"
import SearchResult from "../../components/SearchResult"
import { BodyText, CentredText } from "../../components/Typography"
import ApiManager from "../../lib/ApiManager"

const styles = StyleSheet.create({
  container: {
    paddingBottom: 20,
  },
  subtitle: {
    marginBottom: 10,
  },
})

export class EmptyRoomsScreen extends React.Component {
  static navigationOptions = {
    title: `Empty Rooms`,
  }

  static mapStateToProps = (state) => ({
    token: state.user.token,
  })

  static propTypes = {
    navigation: PropTypes.shape().isRequired,
    token: PropTypes.string,
  }

  static defaultProps = {
    token: ``,
  }

  constructor() {
    super()
    this.state = {
      emptyRooms: [],
      error: null,
      loadingEmptyRooms: true,
      selectedSite: null,
      sites: [],
    }
  }

  componentDidMount() {
    this.fetchEmptyRooms()
  }

  fetchEmptyRooms = async () => {
    const { token } = this.props
    try {
      const emptyRooms = await ApiManager.rooms.getEmptyRooms(token)
      const sites = Array.from(new Set(emptyRooms.map(({ siteid }) => siteid)))
        .map((siteid) => ({
          siteid,
          sitename: emptyRooms.find((r) => r.siteid === siteid).sitename,
        }))
      this.setState({
        emptyRooms,
        loadingEmptyRooms: false,
        sites,
      })
    } catch (error) {
      this.setState({ error: error.message })
    }
  }

  navigateToRoomDetail = (room) => () => {
    const { navigation } = this.props
    navigation.navigate(`RoomDetail`, { room })
  }

  onSelectSite = (value) => {
    this.setState({ selectedSite: value })
  }

  renderEmptyRoom = (room = null) => {
    if (room === null) {
      return null
    }

    const key = `${room.roomid}-${room.roomname}-`
      + `${room.siteid}-${room.classification_name}`

    return (
      <SearchResult
        key={key}
        topText={room.roomname}
        bottomText={room.classification_name}
        type="location"
        buttonText="View"
        onPress={this.navigateToRoomDetail(room)}
      />
    )
  }

  renderResults = () => {
    const { emptyRooms, selectedSite } = this.state
    const matchingRooms = emptyRooms.filter(({ siteid }) => {
      if (selectedSite === null) {
        return true
      }
      return siteid === selectedSite
    })

    if (matchingRooms.length === 0) {
      return (
        <CentredText testID="empty-rooms-message">
          No empty rooms found :(
        </CentredText>
      )
    }
    return matchingRooms.map(this.renderEmptyRoom)
  }

  render() {
    const {
      error,
      sites,
      selectedSite,
      loadingEmptyRooms,
    } = this.state

    if (error) {
      return <BodyText>{`Could not fetch empty rooms: ${error}`}</BodyText>
    }
    if (loadingEmptyRooms) {
      return (
        <Page>
          <View style={styles.container}>
            <ActivityIndicator size="large" />
          </View>
        </Page>
      )
    }
    return (
      <Page>
        <View style={styles.container}>
          <BodyText>
            Filter by:
          </BodyText>
          <Picker
            selectedValue={selectedSite}
            onValueChange={this.onSelectSite}
            testID="building-picker"
          >
            <Picker.Item label="All Buildings" value={null} />
            {
              sites.map(({ sitename, siteid }) => (
                <Picker.Item key={siteid} label={sitename} value={siteid} />
              ))
            }
          </Picker>
          <BodyText style={styles.subtitle}>
            Rooms vacant for the next hour:
          </BodyText>
          {this.renderResults()}
        </View>
      </Page>
    )
  }
}

export default connect(
  EmptyRoomsScreen.mapStateToProps,
  () => ({}),
)(EmptyRoomsScreen)
