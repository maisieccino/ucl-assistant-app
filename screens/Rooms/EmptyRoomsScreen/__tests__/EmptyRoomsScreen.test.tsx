import { cleanup, fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import { waitForEventLoop } from '../../../../jest/test-utils'
import ApiManager from "../../../../lib/ApiManager"
import { EmptyRoomsScreen } from '../EmptyRoomsScreen'

describe(`EmptyRoomsScreen`, () => {
  const mockNavigate = jest.fn()
  const mockProps = {
    navigation: {
      navigate: mockNavigate,
    } as any,
    token: ``,
  }

  const sampleRooms = [
    {
      classification_name: `Classroom`,
      roomname: `Bentham`,
      siteid: `b`,
      sitename: `Bentham House`,
    },
    {
      classification_name: `Lecture Theatre`,
      roomname: `Galactica`,
      siteid: `main`,
      sitename: `Main Building`,
    },
    {
      classification_name: `Classroom`,
      roomname: `Denning`,
      siteid: `main`,
      sitename: `Main Building`,
    },
  ]

  const mockGetEmptyRooms = jest.fn(
    () => Promise.resolve(sampleRooms),
  ) as jest.Mock
  ApiManager.rooms.getEmptyRooms = mockGetEmptyRooms

  beforeAll(() => {
    jest.useFakeTimers()
  })

  beforeEach(() => {
    jest.clearAllMocks()
  })

  afterEach(() => {
    cleanup()
  })

  it(`renders the loading screen`, async () => {
    const wrapper = render(<EmptyRoomsScreen {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it(`calls getEmptyRooms`, async () => {
    const wrapper = render(<EmptyRoomsScreen {...mockProps} />)
    await waitForEventLoop()

    expect(mockGetEmptyRooms).toHaveBeenCalledTimes(1)
    expect(wrapper).toMatchSnapshot()
  })

  it(`shows message when no empty rooms found`, async () => {
    ApiManager.rooms.getEmptyRooms = jest.fn(() => Promise.resolve([]))
    const emptyScreen = render(<EmptyRoomsScreen {...mockProps} />)
    const { getByText } = emptyScreen

    await waitForEventLoop()

    expect(getByText(/empty rooms/i)).toBeTruthy()
    expect(emptyScreen).toMatchSnapshot()

    ApiManager.rooms.getEmptyRooms = mockGetEmptyRooms
  })

  it(`supports filtering by building`, async () => {
    const wrapper = render(<EmptyRoomsScreen {...mockProps} />)
    const { getByTestId } = wrapper
    await waitForEventLoop()

    fireEvent(getByTestId(`building-picker`), `valueChange`, `main`)
    await waitForEventLoop()

    expect(wrapper).toMatchSnapshot()
  })
})
