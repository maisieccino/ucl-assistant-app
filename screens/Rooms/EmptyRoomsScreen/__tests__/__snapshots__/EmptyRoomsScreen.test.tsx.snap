// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyRoomsScreen calls getEmptyRooms 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fefefe",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fefefe",
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingBottom": 60,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            undefined,
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  undefined,
                ]
              }
            >
              Filter by:
            </Text>
            <View>
              <RNCPicker
                items={
                  Array [
                    Object {
                      "label": "All Buildings",
                      "textColor": undefined,
                      "value": null,
                    },
                    Object {
                      "label": "Bentham House",
                      "textColor": undefined,
                      "value": "b",
                    },
                    Object {
                      "label": "Main Building",
                      "textColor": undefined,
                      "value": "main",
                    },
                  ]
                }
                onChange={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="building-picker"
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                  },
                ]
              }
            >
              Rooms vacant for the next hour:
            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "marginBottom": 10,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 5,
                      "padding": 10,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1845,
                      "shadowRadius": 1.62,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    Bentham
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        undefined,
                      ]
                    }
                  >
                    Classroom
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "marginBottom": 10,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 5,
                      "padding": 10,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1845,
                      "shadowRadius": 1.62,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    Galactica
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        undefined,
                      ]
                    }
                  >
                    Lecture Theatre
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "marginBottom": 10,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 5,
                      "padding": 10,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1845,
                      "shadowRadius": 1.62,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    Denning
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        undefined,
                      ]
                    }
                  >
                    Classroom
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`EmptyRoomsScreen renders the loading screen 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fefefe",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fefefe",
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingBottom": 60,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            undefined,
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`EmptyRoomsScreen shows message when no empty rooms found 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fefefe",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fefefe",
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingBottom": 60,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            undefined,
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  undefined,
                ]
              }
            >
              Filter by:
            </Text>
            <View>
              <RNCPicker
                items={
                  Array [
                    Object {
                      "label": "All Buildings",
                      "textColor": undefined,
                      "value": null,
                    },
                  ]
                }
                onChange={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="building-picker"
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                  },
                ]
              }
            >
              Rooms vacant for the next hour:
            </Text>
            <View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#6E707F",
                      "fontFamily": "apercu",
                      "fontSize": 16,
                      "marginBottom": 5,
                      "marginTop": 5,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                No empty rooms found 🙁
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;

exports[`EmptyRoomsScreen supports filtering by building 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fefefe",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#fefefe",
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {
              "paddingBottom": 60,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            undefined,
          ]
        }
        keyboardDismissMode="on-drag"
        refreshControl={null}
        style={
          Array [
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
              "paddingLeft": 0,
              "paddingRight": 0,
            },
            undefined,
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  undefined,
                ]
              }
            >
              Filter by:
            </Text>
            <View>
              <RNCPicker
                items={
                  Array [
                    Object {
                      "label": "All Buildings",
                      "textColor": undefined,
                      "value": null,
                    },
                    Object {
                      "label": "Bentham House",
                      "textColor": undefined,
                      "value": "b",
                    },
                    Object {
                      "label": "Main Building",
                      "textColor": undefined,
                      "value": "main",
                    },
                  ]
                }
                onChange={[Function]}
                selectedIndex={2}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
                testID="building-picker"
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#4D5061",
                    "fontFamily": "apercu",
                    "fontSize": 16,
                    "marginBottom": 1,
                    "marginTop": 1,
                  },
                  Object {
                    "marginBottom": 10,
                  },
                ]
              }
            >
              Rooms vacant for the next hour:
            </Text>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "marginBottom": 10,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 5,
                      "padding": 10,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1845,
                      "shadowRadius": 1.62,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    Galactica
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        undefined,
                      ]
                    }
                  >
                    Lecture Theatre
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "marginBottom": 10,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 5,
                      "padding": 10,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.1845,
                      "shadowRadius": 1.62,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#eaeaea",
                      "borderRadius": 80,
                      "marginRight": 10,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4D5061",
                          "fontFamily": "apercu-bold",
                          "fontSize": 16,
                          "marginBottom": 0,
                          "marginTop": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    Denning
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#6E707F",
                          "fontFamily": "apercu",
                          "fontSize": 16,
                          "fontStyle": "italic",
                          "marginBottom": 1,
                          "marginTop": 0,
                        },
                        undefined,
                      ]
                    }
                  >
                    Classroom
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>,
]
`;
